crates/valence_anvil/src/lib.rs

```diff
@@ -64,18 +100,78 @@ pub struct RegionFolder {
     regions: LruCache<RegionPos, RegionEntry>,
     /// Path to the "region" subdirectory in the world root.
     region_root: PathBuf,
-    /// Scratch buffer for decompression.
-    decompress_buf: Vec<u8>,
+    /// Scratch buffer for (de)compression.
+    compression_buf: Vec<u8>,
+    #[cfg(feature = "write")]
+    write_options: WriteOptions,
 }
 
 impl RegionFolder {
     pub fn new(region_root: impl Into<PathBuf>) -> Self {
         Self {
             regions: LruCache::new(LRU_CACHE_SIZE),
             region_root: region_root.into(),
-            decompress_buf: Vec::new(),
+            compression_buf: Vec::new(),
+            #[cfg(feature = "write")]
+            write_options: WriteOptions::default(),
         }
     }
+
+    /// Set the compression method used to write chunks. This can be useful to
+    /// change in order to write anvil files compatible with older Minecraft
+    /// versions.
+    #[cfg(feature = "write")]
+    pub fn set_write_compression(&mut self, compression: Compression) {
+        self.write_options.compression = compression;
+    }
+
+    /// Set whether to skip writing oversized chunks (>1MiB after compression).
+    /// Versions older than 1.15 (19w36a) cannot read oversized chunks, so this
+    /// may be useful for writing region files compatible with those
+    /// versions.
+    #[cfg(feature = "write")]
+    pub fn skip_write_oversized_chunks(&mut self, skip_oversized_chunks: bool) {
+        self.write_options.skip_oversized_chunks = skip_oversized_chunks;
+    }
+
+    fn get_region<'a>(
+        regions: &'a mut LruCache<RegionPos, RegionEntry>,
+        region_root: &Path,
+        region_x: i32,
+        region_z: i32,
+    ) -> Result<Option<&'a mut Region>, RegionError> {
+        // Need to double get the entry from the cache to make the borrow checker happy.
+        // Polonius will fix this eventually.
+        if regions.get_mut(&(region_x, region_z)).is_some() {
```

Comment:
am I hallucinating, or does this `if` achieve nothing?
(assuming the `None` arm get changes to a `()` value)

also, why `get_mut`?

Reply:
It achieves making the borrow checker happy.